xarray/backends/zarr.py

```diff
@@ -1197,6 +1297,22 @@ def open_dataset(  # type: ignore[override]  # allow LSP violation, not supporti
     def open_datatree(
         self,
         filename_or_obj: str | os.PathLike[Any] | BufferedIOBase | AbstractDataStore,
+        *,
+        mask_and_scale=True,
+        decode_times=True,
+        concat_characters=True,
+        decode_coords=True,
+        drop_variables: str | Iterable[str] | None = None,
+        use_cftime=None,
+        decode_timedelta=None,
+        group=None,
```

Comment:
Thanks, @keewis, for bringing this up. We added the group parameter in case someone decides just to open a specific group within the datatree (e.g. `open_datatree(path2tree, group='group_0/subgroup_1'`). I will define this as `str | Iterable[str] | callable`

Reply:
if it turns out to be too difficult, we can also aim for `group: str | None = None`, and add `Iterable[str] | Callable` in a later PR (we would need additional code and tests for that new feature)

Reply:
I got a Mypy error. I am just rolling back to the original idea and then we can add a new PR in the future.

Reply:
@keewis I will retry with the open_datatree function for h5 files to see if it works. I feel like I tested it using `None`, but I'm not sure. Let me see if I was making a mistake.

Reply:
I ran the following command locally to see what was happening 

```python
 python -m mypy --follow-imports=skip xarray/backends/h5netcdf_.py 
```
and this is the output 

```python 
xarray/backends/h5netcdf_.py:477: error: Value of type "str | Iterable[str] | Callable[..., Any] | None" is not indexable  [index] 
Found 2 errors in 1 file (checked 1 source file)
```
seems like something is happening in this line

```python
group_path = str(parent / group[1:])
```
I think it is because I am using  `group[1:]`  slicing. I need to concatenate the parent node with the subgroup. This subgroup is a string that starts with `/` (e.g. '/subgroup_1'), so if I use the whole string, it won't allow string concatenation. Any thoughts how to handle this?

Reply:
> open_datatree function for h5 files to see if it works

Remember, these are actually still netcdf4 files, just using a different library to access. <https://github.com/aladinor/xarray/blob/datatree-zarr/xarray/backends/h5netcdf_.py#L157-L164>

I'm still looking to see if the open_datatree can be simplified, trying to understand where all of the keywords came from.

Reply:
@keewis and @flamingbear, I think I solved the issue with type hints. It was caused by my using `group[1:]` slicing notation. I refactored it and submitted a new commit.